datamade/django-salsa-auth
# Use Mailchimp api

## Overview
This takes the same idea of our Salsa signup/login process here and adjusts it for the Mailchimp api instead.

### Notes
The file `mailchimp.py` is displaying here as being a new file, but it's just a renamed and adjusted `salsa.py`.

Also I'm getting an error when bringing up the container that there's `No module named 'mailchimp_auth'` but then it proceeds to finish bringing up the container and everything in the app works just fine. I imagine this temporary error is because of the order in which things are installed?

```
bga-payroll-worker    | Traceback (most recent call last):
bga-payroll-worker    |   File "/usr/local/bin/celery", line 8, in <module>
bga-payroll-worker    |     sys.exit(main())
bga-payroll-worker    |   File "/usr/local/lib/python3.5/site-packages/celery/__main__.py", line 16, in main
bga-payroll-worker    |     _main()
bga-payroll-worker    |   File "/usr/local/lib/python3.5/site-packages/celery/bin/celery.py", line 322, in main
bga-payroll-worker    |     cmd.execute_from_commandline(argv)
... extra lines from celery ...
bga-payroll-worker    |   File "/usr/local/lib/python3.5/site-packages/celery/fixups/django.py", line 116, in django_setup
bga-payroll-worker    |     django.setup()
bga-payroll-worker    |   File "/usr/local/lib/python3.5/site-packages/django/__init__.py", line 24, in setup
bga-payroll-worker    |     apps.populate(settings.INSTALLED_APPS)
bga-payroll-worker    |   File "/usr/local/lib/python3.5/site-packages/django/apps/registry.py", line 91, in populate
bga-payroll-worker    |     app_config = AppConfig.create(entry)
bga-payroll-worker    |   File "/usr/local/lib/python3.5/site-packages/django/apps/config.py", line 90, in create
bga-payroll-worker    |     module = import_module(entry)
bga-payroll-worker    |   File "/usr/local/lib/python3.5/importlib/__init__.py", line 126, in import_module
bga-payroll-worker    |     return _bootstrap._gcd_import(name[level:], package, level)
bga-payroll-worker    |   File "<frozen importlib._bootstrap>", line 985, in _gcd_import
bga-payroll-worker    |   File "<frozen importlib._bootstrap>", line 968, in _find_and_load
bga-payroll-worker    |   File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
bga-payroll-worker    | ImportError: No module named 'mailchimp_auth'
bga-payroll-worker    | Signal handler <promise@0x4004a063d8 --> <bound method Celery._autodiscover_tasks of <Celery bga_database at 0x4004926e48>>> raised: AppRegistryNotReady("Apps aren't loaded yet.",)
```

## Testing Instructions
- Set this up with [the mailchimp branch of bga-payroll](https://github.com/datamade/bga-payroll/tree/mailchimp-prep):
  - Follow directions found in that repo's readme if needed
  - Add the local directory path for this repo to the app volumes in `docker-compose.yml` in the format `- <your path>:/django-salsa-auth`
  - Bring up the payroll container
- Confirm that the following processes work as expected:
  - Logging in with a non-subscribed user and getting an error message displayed
  - Signing up and subscribing a new user
    - Before going through the email verification, it may be worth trying to log in with that user to confirm that the "We've already sent an email..." message displays as expected
  - Logging in with a subscribed user and searching
